app/code/community/Apruve/ApruvePayment/controllers/WebhookController.php

```diff
@@ -118,39 +131,52 @@ protected function _changeOrderStatus($orderId)
      * Change the order status based on the order increment ID
      *
      * @param string $orderId
+     *
      * @return bool
      */
     protected function _paymentTermAccepted($orderId)
     {
-        $order = Mage::getModel('sales/order')->loadByIncrementId($orderId);
-        Mage::helper('apruvepayment')->logException($order->getData());
-        Mage::helper('apruvepayment')->logException($orderId);
+        try {
+            $order = Mage::getModel('sales/order')->loadByIncrementId($orderId);
 
-        if ($order) {
-            Mage::helper('apruvepayment')->logException('creating payment accepted...');
+            if ($order) {
+                Mage::helper('apruvepayment')->logException('creating payment accepted...');
 
-            $order->setStatus('buyer_approved');
+                $order->setStatus('buyer_approved');
 
-            $order->save();
-            return true;
+                $order->save();
+
+                return true;
+            }
+        } catch (Exception $e) {
+            Mage::log("Cannot find this order in Magento - possible duplicate webhook - paymentTermAccepted - OrderId: {$orderId}");
+            Mage::log("Message: {$e->getMessage()}");
         }
 
+
         return false;
     }
 
     /**Ball
      * Change the order status based on the order increment ID
      *
      * @param string $orderId
+     *
      * @return bool
      */
     protected function _createInvoice($orderId)
     {
-        if ($orderId) {
-            /** @var Mage_Sales_Model_Order_Invoice_Api $iApi */
-            $iApi = Mage::getModel('sales/order_invoice_api');
-            $invoiceId = $iApi->create($orderId, array());
-            return true;
+        try {
+            if ($orderId) {
+                /** @var Mage_Sales_Model_Order_Invoice_Api $iApi */
+                $iApi      = Mage::getModel('sales/order_invoice_api');
+                $invoiceId = $iApi->create($orderId, array());
+
+                return true;
+            }
+        } catch (Exception $e) {
```

Comment:
I think this is calling an Apruve API, not listening for a webhook, correct?

Reply:
Check out https://github.com/apruve/apruve-magento/blob/fdb9afe8a9052910858ebaf0a31f2389c8ce38eb/app/code/community/Apruve/ApruvePayment/controllers/WebhookController.php#L42

You can see that these events are based upon Apruve webhooks incoming to Magento.